opennebula/shared_schemas.go

```diff
@@ -86,6 +87,38 @@ func diskFields(customFields ...map[string]*schema.Schema) map[string]*schema.Sc
 			Type:     schema.TypeString,
 			Optional: true,
 		},
+		"type": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Type of the disk: swap or fs.",
+			ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+				validtypes := []string{"swap", "fs"}
+				value := v.(string)
+
+				if inArray(value, validtypes) < 0 {
+					errors = append(errors, fmt.Errorf("Type %q must be one of: %s", k, strings.Join(validtypes, ",")))
+				}
+
+				return
+			},
+			ConflictsWith: []string{"image_id"},
```

Comment:
So you also must add for `image_id` a line with `ConflictsWith: []string{"type", "format"}` too